### PR TITLE
Rank changes, Warrant Officers, and Smartgun Sprites

### PR DESCRIPTION
**Pull request by TheGreatRevanchist**
---PRIMARY CHANGES---

--MISC--
Name changes added where necessary (For the Deck Officer, I believe exclusively)

--RANKS--
Added "Warrant Officers" as an between of enlisted and officers, rank insignias included

-COMMAND-
CMO now able to play from O4 to O5 Fleet and Marines

Deck Chief renamed to Deck Officer, may now able to play from W2 TO W3 and O2 to O3

-SENIOR/SPECIALIST ROLES-
Senior Engineer now able to play from E7 to E8 and W1 to W3
Physician now able to play from O3 to O4
Psychiatrist now able to play from O2 to O3
Brig Chief now able to play from E6 to E7 and W1 to W3
Chaplain now able to play from O2 to O3
Pilot now able to play from E5 to E7 and W1 to W2

--SPRITES--

-Smartgun-
Modified smartgun to reflect a more "Streamlined" look, see https://imgur.com/a/Tzzh9FF

### CODE REVIEW

File: modular_mithra/code/modules/mob/inventory.dm [hunk 1]
```diff
@@ -10,7 +10,7 @@
 			return TRUE // self destroying objects (tk, grabs)
 
 		if(W.loc != target)
-			addtimer(CALLBACK(W, /atom/movable/.proc/do_putdown_animation, target, src), 0)
+			INVOKE_ASYNC(W, /atom/movable/.proc/do_putdown_animation, target, src)
```
Comment (Roland410) [inline modular_mithra/code/modules/mob/inventory.dm:13]: This shouldn't be here.

**Comment by Roland410** [comment]
Also if you're renaming DC, may want to give the BC the same treatment.

**Comment by Higgin** [comment]
Love the access changes, don't mind the rank changes, dislike the smartgun sprite.

It looks very flat with the current color palette - especially compared to the stark colors on the current smartgun/e-gun sprites. The shape itself is heavily-reminiscent of the Smart Service Revolver and stun revolver. I'd just as soon you make a non-officer version of the SSR or add this as an alternative sprite option for security equipment kits rather than get rid of the current sprites.

**Comment by Andromeda-K22** [comment]
Atomize the smartgun changes, please.

**Comment by BurgerLUA** [comment]
Conflicts.

**Pull request closed without merging**